plugins/ocm-backend/src/helpers/parser.ts

```diff
@@ -58,3 +60,41 @@ export const parseManagedCluster = (cluster: any): ClusterDetails => {
     ...parsedClusterInfo,
   };
 };
+
+export const parseUpdateInfo = (clusterInfo: any) => {
+  const versionAvailableUpdates =
+    clusterInfo.status.distributionInfo.ocp.versionAvailableUpdates;
+  const versionsAvailable =
+    clusterInfo.status.distributionInfo.ocp.availableUpdates;
+  const version = versionsAvailable
+    ? maxSatisfying(versionsAvailable, '*')
+    : undefined;
+  const url = versionAvailableUpdates
+    ? versionAvailableUpdates[versionsAvailable.indexOf(version)].url
+    : undefined;
+
+  return {
+    update: {
+      available: versionsAvailable ? versionsAvailable.length !== 0 : false,
+      version: version,
+      url: url,
+    },
+  };
+};
+
+export const getClusterInfoByName = (
+  clusterInfos: Array<any>,
+  clusterName: string,
+  logger: Logger,
+) => {
+  const clusterInfo = clusterInfos.filter(
+    (cluster: any) => cluster.metadata.name === clusterName,
```

Comment:
There is no way there are multiple kubernetes resources with the same name. This is not allowed by design. You can use `.find()` instead of filter directly.

Reply:
Resolved in the lastest commit.